instrumentation/events/lttng-module/net.h

```diff
@@ -8,7 +8,188 @@
 #include <linux/skbuff.h>
 #include <linux/netdevice.h>
 #include <linux/ip.h>
+#include <linux/ipv6.h>
+#include <linux/tcp.h>
 #include <linux/version.h>
+#include <lttng-endian.h>
+#include <net/sock.h>
+
+#ifndef ONCE_LTTNG_NET_H
+#define ONCE_LTTNG_NET_H
+
+static inline unsigned char __has_network_hdr(struct sk_buff *skb)
+{
+	/* If the header is not set yet,
+	 * the network header will point to the head. */
+	return skb_network_header(skb) != skb->head;
+}
+
+static struct lttng_event_field emptyfields[] = {
+};
+
+/* Structures for network headers. */
+
+static struct lttng_event_field ipv4fields[] = {
+	[0] = {
+		.name = "version",
+		.type = __type_integer(uint8_t, 4, 4, 0, __BIG_ENDIAN, 10, none),
+	},
+	[1] = {
+		.name = "ihl",
+		.type = __type_integer(uint8_t, 4, 4, 0, __BIG_ENDIAN, 10, none),
+	},
+	[2] = {
+		.name = "tos",
+		.type = __type_integer(uint8_t, 0, 0, 0, __BIG_ENDIAN, 10, none),
+	},
+	[3] = {
+		.name = "tot_len",
+		.type = __type_integer(uint16_t, 0, 0, 0, __BIG_ENDIAN, 10, none),
+	},
+	[4] = {
+		.name = "id",
+		.type = __type_integer(uint16_t, 0, 0, 0, __BIG_ENDIAN, 16, none),
+	},
+	[5] = {
+		.name = "frag_off",
+		.type = __type_integer(uint16_t, 0, 0, 0, __BIG_ENDIAN, 10, none),
+	},
+	[6] = {
+		.name = "ttl",
+		.type = __type_integer(uint8_t, 0, 0, 0, __BIG_ENDIAN, 10, none),
+	},
+	[7] = {
+		.name = "protocol",
+		.type = __type_integer(uint8_t, 0, 0, 0, __BIG_ENDIAN, 16, none),
+	},
+	[8] = {
+		.name = "checksum",
+		.type = __type_integer(uint16_t, 0, 0, 0, __BIG_ENDIAN, 16, none),
+	},
+	[9] = {
+		.name = "saddr",
+		.type = {
+			.atype = atype_array,
+			.u.array.elem_type = __type_integer(uint8_t, 0, 0, 0, __BIG_ENDIAN, 10, none),
```

Comment:
It is within a structure, which needs to be aligned on the largest alignment of the fields contained therein. Then we need to calculate the offset in the structure at which this alignment appears. If this offset is already a multiple of the alignment (8-bit in this case), then there will never be any padding added.